test/e2e/common_test.go

```diff
@@ -2,35 +2,56 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-package e2e
+package e2e_test
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
 	"io"
 	"os"
 	"strings"
-	"time"
 
 	. "github.com/onsi/gomega"
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
-	"knative.dev/pkg/apis"
 
 	corev1 "k8s.io/api/core/v1"
-	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
-	utilrand "k8s.io/apimachinery/pkg/util/rand"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/util/rand"
+	knativeapis "knative.dev/pkg/apis"
 
-	operator "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/test/integration/utils"
 )
 
+func generateTestID(id string) string {
+	return id + "-" + rand.String(5)
+}
+
+func removeTestIDSuffix(id string) string {
+	return id[:len(id)-6]
+}
+
+func createBuild(testBuild *utils.TestBuild, identifier string, filePath string) *buildv1alpha1.Build {
+	build, err := buildTestData(testBuild.Namespace, identifier, filePath)
+	Expect(err).ToNot(HaveOccurred(), "Error retrieving build test data")
+
+	amendBuild(identifier, build)
+
+	err = testBuild.CreateBuild(build)
+	Expect(err).ToNot(HaveOccurred(), "Unable to create build %s", identifier)
+	Logf("Build %s created", identifier)
+
+	build, err = testBuild.GetBuildTillValidation(build.Name)
```

Comment:
`GetBuildTillValidation` does not make use of the retry-loop for the client.get calls. I like your code in lookup_test with the polling in the `lookupRuntimeObject` function. Maybe we should now take the chance to move this polling logic for get calls over to the util functions in the integration tests. We see errors like the `etcdserver: request timed out` often enough to also have their the retries. What do you think ?

Reply:
Good point, let me check how to do that.

Reply:
Done